src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -874,6 +864,241 @@ TEST_F(TestCASESession, Sigma1ParsingTest)
+    encodeParams.msgR2Encrypted.Alloc(encodeParams.encrypted2Length);
+
+    // responder Session Parameters
+    ReliableMessageProtocolConfig MRPConfig = GetDefaultMRPConfig();
+    encodeParams.responderMrpConfig         = &MRPConfig;
+
+    {
+        System::PacketBufferHandle msg;
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma2(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle nonEmptyMsg = System::PacketBufferHandle::New(100);
+
+        // EncodeSigma2 should fail when the packetBufferHandle passed to it is not empty
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma2(nonEmptyMsg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg;
+
+        // EncodeSigma2 should fail when there is no public key
+        encodeParams.pEphPubKey = nullptr;
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma2(msg, encodeParams));
+    }
+
+    encodeParams.pEphPubKey = &EphemeralKey->Pubkey();
+
+    {
+        System::PacketBufferHandle msg;
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma2(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma2 should fail when TBEData2Encrypted is not allocated
+        encodeParams.msgR2Encrypted.Free();
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma2(msg, encodeParams));
+    }
+
+    encodeParams.msgR2Encrypted.Alloc(encodeParams.encrypted2Length);
+    {
+        System::PacketBufferHandle msg;
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma2(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma2 should fail when the encrypted2Length is not set
+        encodeParams.encrypted2Length = 0;
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma2(msg, encodeParams));
+    }
+    // Set encrypted2Length again
+    encodeParams.encrypted2Length = kEncrypted2datalen + CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES;
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma2 should fail when MRP config is missing
+        encodeParams.responderMrpConfig = nullptr;
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma2(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg;
+        // Succeed when MRP Config is provided
+        encodeParams.responderMrpConfig = &MRPConfig;
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma2(msg, encodeParams));
+    }
+    // Release EphemeralKeyPair
+    gDeviceOperationalKeystore.ReleaseEphemeralKeypair(EphemeralKey);
+}
+
+TEST_F(TestCASESession, EncodeSigma2ResumeTest)
+{
+    CASESession::EncodeSigma2ResInputs encodeParams;
+
+    encodeParams.responderSessionId = 7315;
+
+    // responder Session Parameters
+    ReliableMessageProtocolConfig MRPConfig = GetDefaultMRPConfig();
+    encodeParams.responderMrpConfig         = &MRPConfig;
+
+    {
+        System::PacketBufferHandle msg;
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma2Resume(msg, encodeParams));
```

Comment:
Again: static if it does not use member state, use a clear CASESession for each call if it does.